### ISSUE TITLE
Allow common types in entity parent-types lists and action `appliesTo`

### ISSUE THREAD

**Issue by cdisselkoen** [cedar-policy/cedar#1064]
### Category

Cedar validation features/changes

### Describe the feature you'd like to request

_From [this discussion](https://github.com/cedar-policy/cedar/pull/1060#discussion_r1674606979) on #1060._

Today, in entity parent-types lists and action `appliesTo` specs, both schema formats only accept entity type names, not common types that resolve to entity types (or common types that resolve to common types that resolve to entity types...).  This feature request is to allow common types in both of these positions, as long as they (in the end) resolve to entity types.

This needs to be designed/implemented carefully because of the potential for a breaking change. If there is a common and entity type with the same name (in the same namespace, after #1060), and if you write that name in an entity parent-type list or an action `appliesTo` spec, today you get the entity type.  With a naive implementation of this feature request, you might instead get the common type, which would be breaking.  The common type could be not-an-entity-type, in which case a previously valid schema now gives an error, or (even worse, potentially), the common type could be a different entity type, in which case your schema silently now means something else.  To avoid this, maybe entity types should be prioritized over common types in these specific positions, even though RFC 24 specifies that common types are prioritized over entity types in general in the human schema syntax.  (There is less ambiguity in the JSON syntax, which syntactically distinguishes references-to-entity-types from references-to-common-types, although of course it doesn't currently distinguish those in these positions because it doesn't allow common types in these positions.)

### Describe alternatives you've considered

The status quo is to not allow common types, which is also acceptable, if slightly disappointing.

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request
- [ ] ⚠️ This feature might incur a breaking change

**Comment by khieta** [comment]
After offline discussion, the Cedar team decided that this is potentially useful, but not a priority, so I'll be marking this as backlog.

If this feature would be useful for you, please leave a comment!
